### DISCUSSION TITLE
NumericUpDown when blank = System.NullReferenceException: 'Object reference not set to an instance of an object.'

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#13403] [Q&A]
I realize after upgrading from av 0.10 to 11 all my numeric inputs show the message `System.NullReferenceException: 'Object reference not set to an instance of an object.' when we erase input and leave blank, any number is ok but blank trigger the exception.

I placed a breakpoint on the property setter but it never reaches:
![image](https://github.com/AvaloniaUI/Avalonia/assets/113281/66451c5e-2f77-4d77-a1d9-72901a82c08d)

The exception is catch here:
![image](https://github.com/AvaloniaUI/Avalonia/assets/113281/247454d3-8ac0-4e25-bf59-df725ea26fa2)

Looking for advice on this problem

**Comment by IanRawley** [comment]
This is an error in the binding system where it's trying to push a null from the NumericUpDown to a non-nullable property on the VM.  There are two easy ways to handle it:

1.  Make the property on the VM nullable and add null checks where appropriate.
2. Add a converter to the Binding which blocks nulls.  Something like the following:

```
internal class NullBlocker : IValueConverter
{
    public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        return value;
    }

    public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        return value != null ? value : BindingOperations.DoNothing;
    }
}
```

**Comment by maxkatz6** [comment]
It can't set empty value to an integer, it expects nullable for this scenario. Or you can use a converter.

**Comment by thevortexcloud** [comment]
See this issue for some discussions about this
https://github.com/AvaloniaUI/Avalonia/issues/10793
